modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

```diff
@@ -507,6 +504,36 @@ class PodSpecBuilder {
 
     }
 
+    @PackageScope
+    @CompileDynamic
```

Comment:
Unrelated question: 

In which context is the use of these annotation recommended? 

Generally I have only seen the use of `CompileStatic` but curious about these :)

Reply:
https://groovy-lang.org/metaprogramming.html#xform-CompileStatic
https://groovy-lang.org/metaprogramming.html#xform-PackageScope

😎

Reply:
Abhinav, in this specific case, I used `CompileDynamic` because nested maps are difficult to use under static compilation. There might be some best practice that I don't know about, but it would probably require lots of type annotations, so it was easier to just use dynamic compilation. But in general we try to compile static as much as we can for better performance.

Reply:
Thanks Paolo and Ben - interesting topic

Is there a way we could profile the overall time taken for the various DSL level transformations?

Reply:
If you enable the logging using `nextflow -log trace run .. etc` you should have some time reported in the log file